pkg/transport/generic/generic_transport_controller.go

```diff
@@ -1163,26 +1188,97 @@ func (c *genericTransportController) createOrUpdateWrappedObject(ctx context.Con
 		} else {
 			logger.V(2).Info("Created wrapped object in ITS", "namespace", namespace, "objectName", wrappedObject.GetName(), "resourceVersion", wrappedObject2.GetResourceVersion())
 		}
-		return nil
-	}
-	// if we reached here object already exists, try update object
-	wrappedObject.SetResourceVersion(existingWrappedObject.GetResourceVersion())
-	wrappedObject.SetFinalizers(existingWrappedObject.GetFinalizers())
-	wrappedObject2, err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Update(ctx, wrappedObject, metav1.UpdateOptions{
-		FieldManager: ControllerName,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to update wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
-	}
-	if hi := logger.V(3); hi.Enabled() {
-		hi.Info("Updated wrapped object in ITS", "namespace", namespace, "objectName", wrappedObject.GetName(), "wrappedObject", wrappedObject2)
 	} else {
-		logger.V(2).Info("Updated wrapped object in ITS", "namespace", namespace, "objectName", wrappedObject.GetName(), "wrappedObject", wrappedObject, "resourceVersion", wrappedObject2.GetResourceVersion())
+		// if we reached here object already exists, try update object
+		operation = "update"
+		wrappedObject.SetResourceVersion(existingWrappedObject.GetResourceVersion())
+		wrappedObject.SetFinalizers(existingWrappedObject.GetFinalizers())
+		wrappedObject2, err := c.transportClient.Resource(c.wrappedObjectGVR).Namespace(namespace).Update(ctx, wrappedObject, metav1.UpdateOptions{
+			FieldManager: ControllerName,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to update wrapped object '%s' in destination WEC mailbox namespace '%s' - %w", wrappedObject.GetName(), namespace, err)
+		}
+		if hi := logger.V(3); hi.Enabled() {
+			hi.Info("Updated wrapped object in ITS", "namespace", namespace, "objectName", wrappedObject.GetName(), "wrappedObject", wrappedObject2)
+		} else {
+			logger.V(2).Info("Updated wrapped object in ITS", "namespace", namespace, "objectName", wrappedObject.GetName(), "wrappedObject", wrappedObject, "resourceVersion", wrappedObject2.GetResourceVersion())
+		}
+	}
+
+	// Record sync duration and object size metrics for each object type in the wrapped object
+	syncDuration := time.Since(startTime).Seconds()
+	for _, objInfo := range objectInfos {
+		// Record sync duration metric
+		c.objectSyncDurationHist.WithLabelValues(objInfo.GVK.Group, objInfo.GVK.Version, objInfo.GVK.Kind, operation, namespace).Observe(syncDuration)
+
+		// Record object size metric
+		c.objectSizeHist.WithLabelValues(objInfo.GVK.Group, objInfo.GVK.Version, objInfo.GVK.Kind, namespace).Observe(float64(objInfo.Size))
 	}
 
 	return nil
 }
 
+// extractObjectTypesFromWrapped extracts GVK and size information from objects contained in a wrapped object
+func (c *genericTransportController) extractObjectTypesFromWrapped(wrappedObject *unstructured.Unstructured) []objectInfo {
```

Comment:
Object sizes are already being computed in genericTransportController::wrap. Do not serialize again, that is expensive.